tests/sender/send_data.py

```diff
@@ -36,6 +41,20 @@ def setUp(self):
         # change this value if you want to send another number of test string
         self.default_numbers_sendings = 10
 
+    def wait_for_ready_server(self, address, port):
+        num_tries = 3
+
+        while num_tries > 0:
+            try:
+                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                sock.settimeout(3)
+                sock.connect((address, port))
+                sock.close()
+                break
+            except socket.error:
+                num_tries -= 1
+                time.sleep(1)
+
```

Comment:
**issue(blocking):** I do not understand the reasoning behind this new function and I believe it should be directly removed:

1. What is this function's purpose? Why would we need to wait for the server to be ready? At least for the SSLServer case, [it tells you when it is ready](https://github.com/DevoInc/python-sdk/blob/master/tests/sender/local_servers.py#L59).
2. I suspect you created this because some tests were not working in your environment without this, but all these tests we already working without any issue for the rest, and you have not introduced any change in the SUT. Why is this suddenly needed?
3. I see you have implemented this function twice (one here and another in `tests/sender/send_lookup.py`) and call it in every existing test case.
    1. If this has something to do with the testing server/s, then the correct place for this code would be along with the code that creates that/those servers.
    2. The fact that you need to introduce extra code not used for testing the SUT for your tests to run indicates a code smell and I believe this is the case.

Reply:
1. This was introduced because of the test splitting by module. The SSL server is launched in the runtest.py script and it is not available immediately as it runs asynchronously. From the outside, the test only has an address and a port to check.